components/sidebar/sidebar_channel/channel_pencil_icon/index.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+
+import {getCurrentChannelId} from 'mattermost-redux/selectors/entities/channels';
+import {Channel} from 'mattermost-redux/types/channels';
+
+import {getPostDraft} from 'selectors/rhs';
+import {StoragePrefixes} from 'utils/constants';
+import {GlobalState} from 'types/store';
+
+import ChannelPencilIcon from './channel_pencil_icon';
+
+type OwnProps = {
+    channel?: Channel;
```

Comment:
Seems like we only need the channelID. Maybe we should just pass the id to avoid unneeded rerenders?